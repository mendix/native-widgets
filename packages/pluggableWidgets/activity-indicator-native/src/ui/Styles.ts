import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface ActivityIndicatorStyle extends Style {
    container: ViewStyle;
    indicator: {
        size: "large" | "small";
        color: string;
    };
}

export const defaultActivityStyle: ActivityIndicatorStyle = {
    container: {},
    indicator: {
        size: "large",
        color: "gray"
    }
};
