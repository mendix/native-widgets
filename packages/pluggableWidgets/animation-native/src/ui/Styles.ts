import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface AnimationStyle extends Style {
    container: ViewStyle;
}

export const defaultAnimationStyle: AnimationStyle = {
    container: {
        // All ViewStyle properties are allowed
    }
};
