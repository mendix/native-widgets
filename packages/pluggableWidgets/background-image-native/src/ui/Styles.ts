import { ViewStyle, ImageStyle } from "react-native";
import { Style } from "@mendix/piw-native-utils-internal";

interface CustomImageStyle extends ImageStyle {
    svgColor?: string;
}

export interface BackgroundImageStyle extends Style {
    container: ViewStyle;
    image: CustomImageStyle;
}

export const defaultBackgroundImageStyle: BackgroundImageStyle = {
    container: {},
    image: {}
};
