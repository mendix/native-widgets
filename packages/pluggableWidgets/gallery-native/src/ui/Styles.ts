import { ColorValue, TextStyle, ViewStyle } from "react-native";

export interface TouchableStyleProps {
    rippleColor?: ColorValue;
    borderless?: boolean;
    radius?: number;
    foreground?: boolean;
}

export interface TouchableStyle extends ViewStyle, TouchableStyleProps {}

export interface GalleryStyle {
    container?: ViewStyle;
    emptyPlaceholder?: ViewStyle;
    firstItem?: ViewStyle;
    lastItem?: ViewStyle;
    list?: ViewStyle;
    listItem?: ViewStyle;
    loadMoreButtonContainer?: ViewStyle;
    loadMoreButtonPressableContainer?: TouchableStyle;
    loadMoreButtonCaption?: TextStyle;
}

export const defaultGalleryStyle: GalleryStyle = {
    listItem: { flexGrow: 1 },
    loadMoreButtonContainer: {
        alignSelf: "stretch"
    },
    loadMoreButtonPressableContainer: {
        borderWidth: 1,
        borderStyle: "solid",
        rippleColor: "rgba(0, 0, 0, 0.2)",
        borderColor: "#264AE5",
        backgroundColor: "#264AE5",
        alignItems: "center",
        justifyContent: "center",
        borderRadius: 8,

        minWidth: 48,
        minHeight: 48,
        paddingVertical: 8,
        paddingHorizontal: 8
    },
    loadMoreButtonCaption: {
        color: "#FFFFFF",
        fontSize: 12,
        fontWeight: "bold",
        lineHeight: 14
    }
};
