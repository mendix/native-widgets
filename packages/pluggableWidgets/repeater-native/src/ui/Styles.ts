import { Style } from "@mendix/piw-native-utils-internal";
import { ViewStyle } from "react-native";

export interface RepeaterStyle extends Style {
    container: ViewStyle;
}

export const defaultRepeaterStyle: RepeaterStyle = {
    container: {}
};
