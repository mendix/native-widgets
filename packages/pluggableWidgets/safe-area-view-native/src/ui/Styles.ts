import { ViewStyle } from "react-native";
import { Style } from "@mendix/piw-native-utils-internal";

export interface SafeAreaViewStyle extends Style {
    container: ViewStyle;
}

export const defaultSafeAreaViewStyle: SafeAreaViewStyle = {
    container: {
        flex: 1
    }
};
