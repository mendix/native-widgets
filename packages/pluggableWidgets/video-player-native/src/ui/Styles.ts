import { Style } from "@mendix/piw-native-utils-internal";
import {TextStyle, ViewStyle} from "react-native";

export interface VideoStyle extends Style {
    container: ViewStyle;
    indicator: {
        color: string;
    };
    video: ViewStyle;
    errorMessage: TextStyle;
    fullScreenVideoPlayer: ViewStyle;
    controlBtnContainerStyle: ViewStyle;
    fullScreenVideoStyle:  ViewStyle;
    fullScreenActivityIndicatorStyle: ViewStyle;
}

export const defaultVideoStyle: VideoStyle = {
    container: {
        justifyContent: "center",
        alignItems: "center",
        backgroundColor: "black",
    },
    indicator: {
        color: "white"
    },
    video: {
        width: "100%",
        height: undefined
    },
    errorMessage: {
        color: "white"
    },
    fullScreenVideoPlayer: {
        justifyContent: "center",
        backgroundColor: "black",
        margin: 0
    },
    controlBtnContainerStyle: {
        position: "absolute",
        top: "10%",
        right: 10,
        padding: 2,
        backgroundColor: "rgba(0,0,0,0.5)",
        borderRadius: 1,
        justifyContent: "center",
        alignItems: "center"
    },
    fullScreenVideoStyle: {
        width: "100%",
        height: "100%"
    },
    fullScreenActivityIndicatorStyle: {
        position: "absolute",
        alignSelf: "center"
    }
};

