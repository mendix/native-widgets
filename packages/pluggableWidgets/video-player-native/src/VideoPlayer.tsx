import { flattenStyles } from "@mendix/piw-native-utils-internal";
import { createElement, ReactElement, useEffect, useRef, useState, Fragment, useCallback } from "react";
import {
    ActivityIndicator,
    Platform,
    StatusBar,
    Text,
    TouchableOpacity,
    TouchableWithoutFeedback,
    View,
    Appearance,
    Modal,
    NativeModules
} from "react-native";
import Video, { OnProgressData, VideoProperties } from "react-native-video";
import Icon from "react-native-vector-icons/MaterialIcons";
import { VideoPlayerProps } from "../typings/VideoPlayerProps";
import { defaultVideoStyle, VideoStyle } from "./ui/Styles";
import { isAvailable } from "@mendix/piw-utils-internal";
import deepmerge from "deepmerge";

const enum StatusEnum {
    ERROR = "error",
    LOADING = "loading",
    READY = "ready",
    NOT_READY = "not-ready"
}

export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
    const [styles, setStyles] = useState(flattenStyles(defaultVideoStyle, props.style));
    const timeoutRef = useRef<NodeJS.Timeout>();
    const playerRef = useRef<Video>(null);
    const fullScreenPlayerRef = useRef<Video>(null);
    const [status, setStatus] = useState(StatusEnum.NOT_READY);
    const [videoAspectRatio, setVideoAspectRatio] = useState(0);
    const [fullScreen, setFullScreen] = useState(false);
    const [showControls, setShowControls] = useState(false);
    const [currentPlayTime, setCurrentPlayTime] = useState(0);

    useEffect(() => {
        const alteredStyles = deepmerge({}, styles);
        if (props.aspectRatio && videoAspectRatio !== 0) {
            alteredStyles.video.aspectRatio = videoAspectRatio;
            alteredStyles.container.aspectRatio = videoAspectRatio;
        } else if (!props.aspectRatio) {
            alteredStyles.container.aspectRatio = undefined;
            if (alteredStyles.video.width) {
                alteredStyles.container.width = alteredStyles.video.width;
            }
            if (alteredStyles.video.height) {
                alteredStyles.container.height = alteredStyles.video.height;
            }
        }
        setStyles(alteredStyles);
    }, [props.style, props.aspectRatio, videoAspectRatio]);

    const showControlsHandler = useCallback((): void => {
        clearTimeout(timeoutRef.current as NodeJS.Timeout);

        setShowControls(true);
        timeoutRef.current = global.setTimeout(() => setShowControls(false), 5000);
    }, []);

    useEffect(() => {
        if (props.showControls) {
            showControlsHandler();
        }
        return () => {
            clearTimeout(timeoutRef.current as NodeJS.Timeout);
        };
    }, [fullScreen, props.showControls, showControlsHandler]);

    const onVideoPressHandler = useCallback((): void => {
        if (!props.showControls) {
            return;
        }
        if (!showControls) {
            showControlsHandler();
        } else {
            clearTimeout(timeoutRef.current as NodeJS.Timeout);
            setShowControls(false);
        }
    }, [props.showControls, showControls, showControlsHandler]);

    async function fullScreenHandler(isFullScreen: boolean): Promise<void> {
        setFullScreen(isFullScreen);
        const { NavigationBar } = NativeModules;
        if (NavigationBar) {
            const SystemNavigationBar = (await import("react-native-system-navigation-bar")).default;

            if (isFullScreen) {
                SystemNavigationBar.navigationHide();
                StatusBar.setHidden(true);
            } else {
                SystemNavigationBar.navigationShow();
                StatusBar.setHidden(false);
                const isDark = Appearance.getColorScheme() === "dark";
                StatusBar.setBarStyle(isDark ? "light-content" : "dark-content");
            }
        }
    }

    const videoProps: VideoProperties = {
        testID: props.name,
        source: { uri: isAvailable(props.videoUrl) ? props.videoUrl.value : undefined },
        muted: props.muted,
        repeat: props.loop,
        controls: props.showControls,
        onLoadStart: () => setStatus(StatusEnum.LOADING),
        onError: () => setStatus(StatusEnum.ERROR),
        useTextureView: false,
        resizeMode: props.aspectRatio ? "contain" : "stretch",
        onProgress: ({ currentTime }: OnProgressData) => currentTime && setCurrentPlayTime(currentTime)
    };

    const isAndroid = Platform.OS === "android";
    const errorText = status === StatusEnum.ERROR && <Text style={styles.errorMessage}>The video failed to load</Text>;

    return (
        <Fragment>
            {isAndroid && (
                <Modal
                    visible={fullScreen}
                    onRequestClose={() => fullScreenHandler(false)}
                    testID="fullscreen-modal"
                    statusBarTranslucent
                >
                    <View style={styles.fullScreenVideoPlayer}>
                        <TouchableWithoutFeedback onPress={onVideoPressHandler} testID="fullscreen-overlay">
                            <Video
                                {...videoProps}
                                ref={fullScreenPlayerRef}
                                paused={false}
                                onLoad={data => {
                                    setStatus(StatusEnum.READY);
                                    setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
                                    fullScreenPlayerRef.current?.seek(currentPlayTime);
                                }}
                                testID={`${props.name}-fullscreen`}
                                style={styles.fullScreenVideoStyle}
                            />
                        </TouchableWithoutFeedback>
                        {showControls && (
                            <TouchableOpacity
                                style={styles.controlBtnContainerStyle}
                                onPress={() => fullScreenHandler(false)}
                                testID="btn-fullscreen-exit"
                            >
                                <Icon name="fullscreen-exit" color="white" size={22} />
                            </TouchableOpacity>
                        )}
                        {status === StatusEnum.LOADING && (
                            <ActivityIndicator
                                color={styles.indicator.color}
                                size="large"
                                style={styles.fullScreenActivityIndicatorStyle}
                            />
                        )}
                        {errorText}
                    </View>
                </Modal>
            )}
            <View style={styles.container}>
                {status === StatusEnum.LOADING && <ActivityIndicator color={styles.indicator.color} size="large" />}
                {errorText}
                {!fullScreen && (
                    <TouchableWithoutFeedback style={styles.container} onPress={onVideoPressHandler}>
                        <Video
                            {...videoProps}
                            paused={!props.autoStart && !currentPlayTime}
                            onLoad={data => {
                                setStatus(StatusEnum.READY);
                                setVideoAspectRatio(data.naturalSize.width / data.naturalSize.height);
                                playerRef.current?.seek(currentPlayTime);
                            }}
                            ref={playerRef}
                            style={status !== StatusEnum.READY ? { height: 0 } : styles.video}
                        />
                    </TouchableWithoutFeedback>
                )}
                {isAndroid && showControls && (
                    <TouchableOpacity
                        onPress={() => fullScreenHandler(true)}
                        style={styles.controlBtnContainerStyle}
                        testID="btn-fullscreen"
                    >
                        <Icon name="fullscreen" color="white" size={22} />
                    </TouchableOpacity>
                )}
            </View>
        </Fragment>
    );
}
