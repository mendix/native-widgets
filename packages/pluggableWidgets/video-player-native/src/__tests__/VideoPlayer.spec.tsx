import { dynamicValue } from "@mendix/piw-utils-internal";
import { createElement } from "react";
import { Modal, View } from "react-native";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import { VideoProperties } from "react-native-video";

import { VideoPlayer } from "../VideoPlayer";
import { VideoPlayerProps } from "../../typings/VideoPlayerProps";
import { VideoStyle } from "../ui/Styles";

jest.useFakeTimers();
jest.mock("react-native-video", () => "Video");
jest.mock("react-native-system-navigation-bar", () => ({
    navigationHide: jest.fn(),
    navigationShow: jest.fn()
}));

describe("VideoPlayer", () => {
    let defaultProps: VideoPlayerProps<VideoStyle>;
    beforeEach(() => {
        defaultProps = {
            name: "video-player-test",
            style: [],
            videoUrl: dynamicValue("https://mendix.com/video.mp4"),
            autoStart: false,
            muted: false,
            loop: false,
            showControls: true,
            aspectRatio: false
        };
    });

    it("renders a loading indicator", () => {
        const component = render(<VideoPlayer {...defaultProps} />);

        expect(component.toJSON()).toMatchSnapshot();
    });

    it("passes the right props to the video player", () => {
        const component = render(<VideoPlayer {...defaultProps} />);
        const props = component.getByTestId("video-player-test").props as VideoProperties;

        expect(props.source).toEqual({ uri: "https://mendix.com/video.mp4" });
        expect(props.paused).toBe(true);
        expect(props.muted).toBe(false);
        expect(props.repeat).toBe(false);
        expect(props.controls).toBe(true);
        expect(props.style).toEqual({ height: 0 });
    });

    it("hides the loading indicator after load", () => {
        const component = render(<VideoPlayer {...defaultProps} />);

        fireEvent(component.getByTestId("video-player-test"), "load", { naturalSize: { width: 640, height: 360 } });

        expect(component.toJSON()).toMatchSnapshot();
        expect(component.getByTestId("video-player-test").props.style).toEqual({ width: "100%", height: undefined });
    });

    it("shows the loading indicator if the source changes", () => {
        const component = render(<VideoPlayer {...defaultProps} />);

        fireEvent(component.getByTestId("video-player-test"), "load", { naturalSize: { width: 640, height: 360 } });
        fireEvent(component.getByTestId("video-player-test"), "loadStart");

        expect(component.toJSON()).toMatchSnapshot();
        expect(component.getByTestId("video-player-test").props.style).toEqual({ height: 0 });
    });

    it("load a video and calculate the aspect ratio", () => {
        const component = render(<VideoPlayer {...defaultProps} aspectRatio />);

        fireEvent(component.getByTestId("video-player-test"), "load", { naturalSize: { width: 1080, height: 554 } });

        expect(component.toJSON()).toMatchSnapshot();
        expect(component.UNSAFE_getByType(View).props.style).toEqual({
            alignItems: "center",
            aspectRatio: 1.9494584837545126,
            backgroundColor: "black",
            justifyContent: "center"
        });
    });

    it("renders an error", () => {
        const component = render(<VideoPlayer {...defaultProps} />);

        fireEvent(component.getByTestId("video-player-test"), "error");

        expect(component.getByText("The video failed to load")).toBeDefined();
        expect(component.getByText("The video failed to load").props.style).toEqual({ color: "white" });
    });

    describe("VideoPlayerAndroid", () => {
        beforeAll(() => {
            jest.mock("react-native/Libraries/Utilities/Platform", () => ({
                OS: "android",
                select: jest.fn(dict => dict.android)
            }));
        });
        it("render video with controls", () => {
            const component = render(<VideoPlayer {...defaultProps} />);
            expect(component).toMatchSnapshot();
        });
        it("render video without controls if showControls is set to false", () => {
            const component = render(<VideoPlayer {...defaultProps} showControls={false} />);
            expect(component).toMatchSnapshot();
        });

        it("show fullscreen when press fullscreen icon-- android", async () => {
            const component = render(<VideoPlayer {...defaultProps} />);
            const fullScreenBtn = component.getByTestId("btn-fullscreen");

            fireEvent.press(fullScreenBtn);

            await waitFor(() => {
                component.getByTestId("btn-fullscreen-exit");
                const modal = component.UNSAFE_getByType(Modal);
                expect(modal.props.visible).toBe(true);
            });
        });
    });
});
