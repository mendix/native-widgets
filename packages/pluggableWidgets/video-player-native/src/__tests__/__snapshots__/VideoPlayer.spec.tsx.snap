// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer VideoPlayerAndroid render video with controls 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="fullscreen-modal"
    visible={false}
  />,
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": undefined,
        "backgroundColor": "black",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Video
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      controls={true}
      focusable={true}
      muted={false}
      onClick={[Function]}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      paused={true}
      repeat={false}
      resizeMode="stretch"
      source={
        {
          "uri": "https://mendix.com/video.mp4",
        }
      }
      style={
        {
          "height": 0,
        }
      }
      testID="video-player-test"
      useTextureView={false}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "borderRadius": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 2,
          "position": "absolute",
          "right": 10,
          "top": "10%",
        }
      }
      testID="btn-fullscreen"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 22,
            },
            undefined,
            {
              "fontFamily": "MaterialIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`VideoPlayer VideoPlayerAndroid render video without controls if showControls is set to false 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="fullscreen-modal"
    visible={false}
  />,
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": undefined,
        "backgroundColor": "black",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Video
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      controls={false}
      focusable={true}
      muted={false}
      onClick={[Function]}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      paused={true}
      repeat={false}
      resizeMode="stretch"
      source={
        {
          "uri": "https://mendix.com/video.mp4",
        }
      }
      style={
        {
          "height": 0,
        }
      }
      testID="video-player-test"
      useTextureView={false}
    />
  </View>,
]
`;

exports[`VideoPlayer hides the loading indicator after load 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="fullscreen-modal"
    visible={false}
  />,
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": undefined,
        "backgroundColor": "black",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Video
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      controls={true}
      focusable={true}
      muted={false}
      onClick={[Function]}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      paused={true}
      repeat={false}
      resizeMode="stretch"
      source={
        {
          "uri": "https://mendix.com/video.mp4",
        }
      }
      style={
        {
          "height": undefined,
          "width": "100%",
        }
      }
      testID="video-player-test"
      useTextureView={false}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "borderRadius": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 2,
          "position": "absolute",
          "right": 10,
          "top": "10%",
        }
      }
      testID="btn-fullscreen"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 22,
            },
            undefined,
            {
              "fontFamily": "MaterialIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`VideoPlayer load a video and calculate the aspect ratio 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="fullscreen-modal"
    visible={false}
  />,
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": 1.9494584837545126,
        "backgroundColor": "black",
        "justifyContent": "center",
      }
    }
  >
    <Video
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      controls={true}
      focusable={true}
      muted={false}
      onClick={[Function]}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      paused={true}
      repeat={false}
      resizeMode="contain"
      source={
        {
          "uri": "https://mendix.com/video.mp4",
        }
      }
      style={
        {
          "aspectRatio": 1.9494584837545126,
          "height": undefined,
          "width": "100%",
        }
      }
      testID="video-player-test"
      useTextureView={false}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "borderRadius": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 2,
          "position": "absolute",
          "right": 10,
          "top": "10%",
        }
      }
      testID="btn-fullscreen"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 22,
            },
            undefined,
            {
              "fontFamily": "MaterialIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`VideoPlayer renders a loading indicator 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="fullscreen-modal"
    visible={false}
  />,
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": undefined,
        "backgroundColor": "black",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Video
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      controls={true}
      focusable={true}
      muted={false}
      onClick={[Function]}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      paused={true}
      repeat={false}
      resizeMode="stretch"
      source={
        {
          "uri": "https://mendix.com/video.mp4",
        }
      }
      style={
        {
          "height": 0,
        }
      }
      testID="video-player-test"
      useTextureView={false}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "borderRadius": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 2,
          "position": "absolute",
          "right": 10,
          "top": "10%",
        }
      }
      testID="btn-fullscreen"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 22,
            },
            undefined,
            {
              "fontFamily": "MaterialIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;

exports[`VideoPlayer shows the loading indicator if the source changes 1`] = `
[
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    testID="fullscreen-modal"
    visible={false}
  />,
  <View
    style={
      {
        "alignItems": "center",
        "aspectRatio": undefined,
        "backgroundColor": "black",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <ActivityIndicator
      color="white"
      size="large"
    />
    <Video
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      controls={true}
      focusable={true}
      muted={false}
      onClick={[Function]}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      paused={true}
      repeat={false}
      resizeMode="stretch"
      source={
        {
          "uri": "https://mendix.com/video.mp4",
        }
      }
      style={
        {
          "height": 0,
        }
      }
      testID="video-player-test"
      useTextureView={false}
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "borderRadius": 1,
          "justifyContent": "center",
          "opacity": 1,
          "padding": 2,
          "position": "absolute",
          "right": 10,
          "top": "10%",
        }
      }
      testID="btn-fullscreen"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "white",
              "fontSize": 22,
            },
            undefined,
            {
              "fontFamily": "MaterialIcons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;
